### PR TITLE
Add --no-check-features to the supported linker flags.

### PR DESCRIPTION
**Pull request by kripken**
When feature errors happen, the linker suggests
adding that flag.

### CODE REVIEW

**Comment by kripken** [comment]
Thanks @sbc100, I added a separate list + a  test.

File: tests/test_other.py [hunk 1]
```diff
@@ -9795,6 +9795,13 @@ def test_missing_stdlibs(self):
     # error out when linking with these flags.
     run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.cpp'), '-lm', '-ldl', '-lrt', '-lpthread'])
 
+  @no_fastcomp('lld-specific')
+  def test_supported_linker_flags(self):
+    out = run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.cpp'), '-Wl,waka'], stderr=PIPE).stderr
+    self.assertContained('WARNING: ignoring unsupported linker flag: `waka', out)
+    out = run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.cpp'), '-Wl,--no-check-features'], stderr=PIPE).stderr
+    self.assertNotContained('WARNING: ignoring unsupported linker flag', out)
```
Comment (sbc100) [inline tests/test_other.py]: How about just asserting that its empty?

Comment (kripken) [reply tests/test_other.py]: Just asserting it's empty could hit that random windows error on getting that assert error during shutdown.

However, I realize this is that PR I was trying to remember from before, so this part of the test overlaps with the test just added in #9591, so I'll remove it.

Comment (kripken) [reply tests/test_other.py]: No, wait, this part is the necessary part actually, checking the new flag is in fat supported.

I'll remove the redundant test from the other PR.

**Pull request merged**